src/Security/Items.cs

```diff
@@ -237,11 +234,14 @@ internal SecKeyChain (NativeHandle handle)
 				copy.LowlevelSetObject (CFBoolean.TrueHandle, SecItem.ReturnData);
 				var n = SetLimit (copy, max);
 				
-				IntPtr ptr;
-				result = SecItem.SecItemCopyMatching (copy.Handle, out ptr);
+				result = SecItem.SecItemCopyMatching (copy.Handle, out IntPtr ptr);
 				n = null;
 				if (result == SecStatusCode.Success)
-					return NSArray.ArrayFromHandleFunc<SecRecord> (ptr, (element) => new SecRecord (Runtime.GetNSObject<NSMutableDictionary> (element, false)!))!;
+					return CFArray.ArrayFromHandleFunc<SecRecord> (ptr, (element) => {
+						CFObject.CFRetain (element);
+						var dictionary = Runtime.GetNSObject<NSMutableDictionary> (element, true)!;
```

Comment:
The way to do that is to pass `false` to `GetNSObject` (which will then retain the handle), instead of manually calling `CFObject.CFRetain` on the handle.

```suggestion
						var dictionary = Runtime.GetNSObject<NSMutableDictionary> (element, false)!;
```